app/controllers/application_controller.rb

```diff
@@ -23,12 +28,23 @@ class ApplicationController < ActionController::Base
     end
   end
 
-  rescue_from EditionAssertions::AccessError do |e|
-    Rails.logger.warn(e.message)
-    render "documents/forbidden", status: :forbidden, assigns: { edition: e.edition }
-  end
-
   def rendering_context
     request.headers["Content-Publisher-Rendering-Context"] || "application"
   end
+
+  def check_user_access
+    document_id = request.path_parameters[:document]
+    return if document_id.blank?
+
+    access_limit = Edition.includes(:access_limit)
+      .current(document: document_id)
```

Comment:
I'd just use `find_current` rather than this method as the only thing we won't load is the status and I'm not sure we need the extra scope methods.